src/devices/things/ITemperatureSensor.cs

```diff
@@ -0,0 +1,14 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace Iot.Device
+{
+    /// <summary>
+    /// Abstraction for temperature sensor
+    /// </summary>
+    public interface ITemperatureSensor
+    {
+        double Temperature { get; }
```

Comment:
So what is `double`? This needs to be explained thoroughly. Is there an error state for a return value?

Reply:
We're thinking Celsius since this will be the most intuitive for most of the people in the world perhaps except US, other alternative is Kelvin as International System of Units suggest - but that seems that it will be inconvenient for everyone except for physicist.
I'll add summary saying Celsius. (Fahrenheit/Kelvin can be an extension method)

For error - couple of options:
- NaN
- double.Max/double.Min
- InvalidOperationException

Reply:
Error should be `NaN` or an exception.

Reply:
I would go for Celsius as it's really what is used everywhere (except very few countries). 
For error, double.Max is a good way to check esilly in the code. Having try/catch is not the very best way to manage thos kind of readings.
Other option is to have a "ValidData" property. So you can do
```csharp
if (tempClass.IsValid)
{
    Console.WriteLine(tempClass.Temerature)
}
```

Reply:
> For error, double.Max is a good way to check esilly in the code.

Using any finite value as the error case is bound to lead to confusion or errors, especially if there exists a sensor that legitimately returns that value.

Using any infinite value as the error case is slightly better, but can also lead to failures in the case of overflow or underflow.

`NaN` is an explicit value that, outside of very specific domains (such as complex numbers), generally represents an error value and should be the value used in the case of error (if we are not throwing an exception). As from the IEEE 754 specification:
> Quiet NaNs should, by means left to the implementer’s discretion, afford retrospective diagnostic information inherited from invalid or unavailable data and results.
> ...
> Under default exception handling, any operation signaling an invalid operation exception and for which a floating-point result is to be delivered shall deliver a quiet NaN.

The runtime does not support the IEEE exception handling behavior and explicitly disables it, so the effective behavior is that the floating-point result deliver should return `double.NaN`